modules.py

```diff
@@ -288,4 +304,80 @@ def rebin(coords, wavelength, size, detector_scale):
 
     x = (coords[1]-size//2.)*detector_scale/scale_factor+(size//2.)
     y = (coords[0]-size//2.)*detector_scale/scale_factor+(size//2.)
-    return y, x
\ No newline at end of file
+    return y, x
+
+
+class PolyPSF(OpticalArray):
+    """
+    Polychromatic PSF class. Inherits from OpticalArray.
+    """
+    def __init__(self, band, spectral_type='A', size=505, scale=0.01):
+        super(PolyPSF, self).__init__(size, poly=True, scale=scale)
+        self.band = band
+        self.spectral_type = spectral_type
+        self._wavelength_contributions = None
+        self.sed_file = 'sed_%s0_fe0.txt' % spectral_type.lower()
+        self._x = None
+        self._flux = None
+        self.name = 'polychromatic_psf'
+        self._dist = self.dist
+
+    def get_sed(self):
+        """
+        Reads the SED profile from files and get the spectral energy distribution.
+        :return: self._x, self._flux : two lists, wavelength and its associated flux.
+        """
+        if self._x is not None and self._flux is not None:
+            return self._x, self._flux
+        han = open('SED/%s' % self.sed_file)
+        data = han.readlines()
+        han.close()
+
+        x, flux = [], []
+        for i in range(len(data)):
+            xy = data[i].split()
+            if xy:
+                x.append(float(xy[0]))
+                flux.append(float(xy[1]))
+        self._x = x
+        self._flux = flux
+        return
+
+    def wavelength_contributions(self):
+        """
+        Computes the contributions of all 10 evenly-spaced wavelength in the filter band.
+        :return: wavelength_contributions. tuple of lists, containing the 10 wavelength and their relative contribution
+        """
+        if self._wavelength_contributions is not None:
+            return self._wavelength_contributions
+
+        bands = dict(Z=(0.76, 0.977), Y=(0.927, 1.192), J=(1.131, 1.454), H=(1.380, 1.774), F=(1.683, 2.000),
+                     Wide=(0.927, 2.000))
+
+        spectral_interpolation = interp1d(self._x, self._flux, kind='cubic')
+
+        waves = np.linspace(bands[self.band][0], bands[self.band][1], 10)  # Takes 10 wavelengths in band
+        self._wavelength_contributions = [waves, spectral_interpolation(waves)]
+        return
+
+    def create_polychrome(self):
+        tmp = np.zeros((self.array_size, self.array_size))
+        for i, wavel in enumerate(self._wavelength_contributions[0]):
+            pupil = Pupil(wavel)
+            psf = PSF(pupil, self.scale)
+            psf.resize_psf(wavelength=wavel, size=np.shape(psf.a)[0], scale=self.scale)  # scale is supposed to be 0.01
+            # psf.save('polyPSF_%s' % wavel)  # consistency test: will save all the 10 PSFs.
+            psf._a = psf.a * self._wavelength_contributions[1][i]
```

Comment:
Why is this a direct manipulation of psf._a instead of just acting on psf.a ? And why not use:

psf.a *= self.wavelength_contributions[1][i] 

I'm actually a bit confused by the way you are doing loops here.